packages/signers/src/sign-transaction.ts

```diff
@@ -81,19 +77,16 @@ export async function signAndSendTransactionWithSigners<
         abortSignal
     );
 
-    if (sendingSigner) {
-        abortSignal?.throwIfAborted();
-        const [signature] = await sendingSigner.signAndSendTransactions([signedTransaction], { abortSignal });
-        return signature;
+    if (!sendingSigner) {
+        // TODO: Coded error.
+        throw new Error(
+            'No TransactionSendingSigner was identified. Please provide a valid `ITransactionWithSingleSendingSigner` transaction.'
+        );
     }
 
-    if (config.fallbackSender) {
-        assertTransactionIsFullySigned(signedTransaction);
-        return config.fallbackSender(signedTransaction);
-    }
-
-    // TODO: Coded error.
-    throw new Error('No TransactionSendingSigner was identified and no fallback sender was provided.');
+    abortSignal?.throwIfAborted();
```

Comment:
This belongs right after the last async operation, right?

Reply:
This function awaits multiple times so my pattern was:

```ts
abortSignal?.throwIfAborted();
await thingA({ abortSignal });

abortSignal?.throwIfAborted();
await thingB({ abortSignal });

return;
```

Because `thingB` should fail for us if the `abortSignal` is aborted. However, just in case `thingB` is naughty and doesn't fail for us, I've updated it to:

```diff
  abortSignal?.throwIfAborted();
  await thingA({ abortSignal });
  
  abortSignal?.throwIfAborted();
  await thingB({ abortSignal });
+ abortSignal?.throwIfAborted();
  
  return;
```

Let me know what you think of that pattern for future reference so I follow what you're doing throughout the rest of this library.